components/auth/RoleDetailEdit.vue

```diff
@@ -245,7 +243,19 @@ export default {
       this.$set(row, key, value);
     },
     cancel() {
-      this.$router.replace(this.doneLocationOverride);
+      this.done();
+    },
+    async actuallySave(url) {
+      if ( this.isCreate ) {
+        url = url || this.schema.linkFor('collection');
+        const res = await this.value.save({ url, redirectUnauthorized: false });
+
+        if (res) {
+          Object.assign(this.value, res);
```

Comment:
Will this cause problems with reactivity?

We might need to use this.$set.

Reply:
Interesting point! I brought this over from CreateEditView so hoping it's safe or at least not a regression